packages/visx-responsive/src/components/ParentSize.tsx

```diff
@@ -29,74 +31,65 @@ type ParentSizeState = {
 
 export type ParentSizeProvidedProps = ParentSizeState;
 
-export default class ParentSize extends React.Component<
-  ParentSizeProps & Omit<JSX.IntrinsicElements['div'], keyof ParentSizeProps>,
-  ParentSizeState
-> {
-  static defaultProps = {
-    debounceTime: 300,
-    enableDebounceLeadingCall: true,
-    parentSizeStyles: { width: '100%', height: '100%' },
-  };
-  animationFrameID: number = 0;
-  resizeObserver: ResizeObserver | undefined;
-  target: HTMLDivElement | null = null;
+export default function ParentSize({
+  className,
+  children,
+  debounceTime = 300,
+  disableFor,
+  parentSizeStyles = { width: '100%', height: '100%' },
+  enableDebounceLeadingCall = true,
+  ...restProps
+}: ParentSizeProps & Omit<JSX.IntrinsicElements['div'], keyof ParentSizeProps>) {
+  const target = useRef<HTMLDivElement | null>(null);
+  const animationFrameID = useRef(0);
 
-  state = {
-    width: 0,
-    height: 0,
-    top: 0,
-    left: 0,
-  };
+  const [state, setState] = useState<ParentSizeState>({ width: 0, height: 0, top: 0, left: 0 });
 
-  componentDidMount() {
-    this.resizeObserver = new ResizeObserver((entries = [] /** , observer */) => {
+  const resize = useMemo(() => {
+    const normalized = disableFor ? (Array.isArray(disableFor) ? disableFor : [disableFor]) : [];
+
+    return debounce(
+      (incoming: ParentSizeState) => {
+        setState(existing => {
+          const stateKeys = Object.keys(existing) as (keyof ParentSizeState)[];
+          const keysWithChanges = stateKeys.reduce<(keyof ParentSizeState)[]>((diff, key) => {
+            return existing[key] === incoming[key] ? diff : [...diff, key];
+          }, []);
```

Comment:
Could we use `filter` here instead? I think it's a little more readable

```suggestion
          const keysWithChanges = stateKeys.filter(key => existing[key] !== incoming[key])
```

Reply:
Whoops, I agree. I think I had a started with some different data structure and finally settled on this where my reduce end up being an over-engineered filter.